include/boost/geometry/algorithms/detail/overlay/get_turns.hpp

```diff
@@ -519,8 +591,9 @@ template
 >
 struct get_turns_cs
 {
-    typedef typename geometry::point_type<Range>::type point_type;
+    typedef typename geometry::point_type<Range>::type range_point_type;
     typedef typename geometry::point_type<Box>::type box_point_type;
+    typedef boost::array<box_point_type, 4> box_array;
```

Comment:
Boost.Array requires that the stored type has the default constrictor. Just pointing it out, there are probably many default-constructed points in the code. Still AFAIK this requirement is not a part of the Point concept.